lua/smear_cursor/color.lua

```diff
@@ -58,72 +38,95 @@ local function interpolate_colors(hex1, hex2, t)
+
+	local hl_group = ("SmearCursorNormal%s"):format(opts.inverted and "Inverted" or "", opts.level or 0)
+
+	-- Get the cursor color from the treesitter highlight group
+	-- at the cursor.
+	if cursor_color == "treesitter" and vim.b.ts_highlight then
+		local cursor = vim.api.nvim_win_get_cursor(0)
+		local ts_hl_group ---@type string?
+		for _, capture in pairs(vim.treesitter.get_captures_at_pos(0, cursor[1] - 1, cursor[2])) do
+			ts_hl_group = "@" .. capture.capture .. "." .. capture.lang
+		end
+		local ts_color = ts_hl_group and get_hl_color(ts_hl_group, "fg")
+		if ts_color then
+			_cursor_color = ts_color
+			hl_group = hl_group .. "_" .. ts_color:sub(2)
+		end
+	end
+
+	if cache[hl_group] then
+		return hl_group
+	end
+
+	local _normal_bg = normal_bg or get_hl_color("Normal", "bg") or "none"
+
 	-- Retrieve the cursor color and the normal background color if not set by the user
-	local _cursor_color = cursor_color
-		or get_hl_color("Cursor", "background")
-		or get_hl_color("Normal", "foreground")
-		or "#d0d0d0"
-	local _normal_bg = normal_bg or get_hl_color("Normal", "background") or "none"
+	_cursor_color = _cursor_color or get_hl_color("Cursor", "bg") or get_hl_color("Normal", "fg") or "#d0d0d0"
 
 	-- Blending breaks with transparent backgrounds
 	local blending = config.legacy_computing_symbols_support and _normal_bg ~= "none"
 
-	vim.api.nvim_set_hl(0, M.hl_group, {
-		fg = _cursor_color,
-		bg = "none",
-		blend = blending and 100 or 0,
-	})
-	vim.api.nvim_set_hl(
-		0,
-		M.hl_group_inverted,
-		-- Blending does not work as we'd like with reversed colors
-		{ fg = _normal_bg == "none" and transparent_bg_fallback_color or _normal_bg, bg = _cursor_color, blend = 0 }
-	)
-
-	M.hl_groups = {}
-	M.hl_groups_inverted = {}
-
-	for i = 1, config.color_levels do
-		local opacity = (i / config.color_levels) ^ (1 / config.gamma)
-		local blended_cursor_color = interpolate_colors(
+	if opts.level then
+		local opacity = (opts.level / config.color_levels) ^ (1 / config.gamma)
+		_cursor_color = interpolate_colors(
 			_normal_bg == "none" and transparent_bg_fallback_color or _normal_bg,
 			_cursor_color,
 			opacity
 		)
-		local blended_hl_group = M.hl_group .. i
-		local blended_hl_group_inverted = M.hl_group_inverted .. i
-		M.hl_groups[i] = blended_hl_group
-		M.hl_groups_inverted[i] = blended_hl_group_inverted
-
-		vim.api.nvim_set_hl(0, blended_hl_group, {
-			fg = blended_cursor_color,
-			bg = _normal_bg,
-			blend = blending and 100 or 0,
-		})
-		vim.api.nvim_set_hl(0, blended_hl_group_inverted, {
-			fg = _normal_bg == "none" and transparent_bg_fallback_color or _normal_bg,
-			bg = blended_cursor_color,
-			blend = 0,
-		})
 	end
+
+	---@type vim.api.keyset.highlight
+	local hl = opts.inverted
+			and {
+				fg = _normal_bg == "none" and transparent_bg_fallback_color or _normal_bg,
+				bg = _cursor_color,
+				blend = 0,
+			}
+		or {
+			fg = _cursor_color,
+			bg = "none", -- _normal_bg,
```

Comment:
In the old code, for level=0, it used to be "none" and for level>0 it was _normal_bg.
I assume this was a bug? And you always want `none` here?